File: benchmarks/pandas/bench_dataframe_rank.py
"""Benchmark: DataFrame.rank on a 10k-row DataFrame"""
import json, time
import numpy as np
import pandas as pd

ROWS = 10_000
WARMUP = 3
ITERATIONS = 20

a = np.sin(np.arange(ROWS) * 0.1)
b = np.cos(np.arange(ROWS) * 0.1)
df = pd.DataFrame({"a": a, "b": b})

for _ in range(WARMUP):
    df.rank()

start = time.perf_counter()
for _ in range(ITERATIONS):
    df.rank()
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "dataframe_rank",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))

File: benchmarks/tsb/bench_dataframe_rank.ts
/**
 * Benchmark: rankDataFrame on a 10k-row DataFrame
 */
import { DataFrame, rankDataFrame } from "../../src/index.js";

const ROWS = 10_000;
const WARMUP = 3;
const ITERATIONS = 20;

const a = Array.from({ length: ROWS }, (_, i) => Math.sin(i * 0.1));
const b = Array.from({ length: ROWS }, (_, i) => Math.cos(i * 0.1));
const df = DataFrame.fromColumns({ a, b });

for (let i = 0; i < WARMUP; i++) {
  rankDataFrame(df);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  rankDataFrame(df);
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "dataframe_rank",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
